lib/html_cleaner.rb

```diff
@@ -144,9 +146,18 @@ def add_paragraphs_to_text(text)
   # These assume they are running on well-formed XHTML, which we can do
   # because they will only be used on already-cleaned fields.
 
-  # strip img tags
-  def strip_images(value)
-    value.gsub(/<img .*?>/, '')
+  # strip img tags, optionally leaving the src URL exposed
+  def strip_images(value, keep_src: false)
+    value.gsub(%r{(<img .*?>)}) do |img_tag|
+      if keep_src
+        # The cleaned field will always use double quotes (") but the fields
+        # in tests aren't always cleaned, so we're allowing double or single
+        # quotes here.
```

Comment:
Oof, how sure is it that it's only the tests where the fields aren't cleaned? An uncleaned field could have unexpected formatting beyond just different quotes, which would break/bypass the two regular expressions here.

A lot of the time, this function is called from `sanitize_field`. I think it would be more reliable to use Sanitize or nokogiri in `sanitize_value` to do the stripping. There we're already dealing with those libs and I'd expect them to be more reliable than making our own regexp (don't reinvent the wheel and all that).

Doing the image stripping in `sanitize_value` would also make things a bit nicer because we could use a config similar to `ArchiveConfig.FIELDS_ALLOWING_VIDEO_EMBEDS` to designate fields to clean (or rather, not clean), instead of calling strip_images during display. That also makes it less likely that some display spots will get missed like it already happened in AO3-6564.

With the source urls getting preserved, doing this in the sanitisation step also wouldn't lose any data, if that was a concern previously.